File: app/src/main/java/com/crevion/apps/cleanandroidcode/home/HomeView.java
package com.crevion.apps.cleanandroidcode.home;

import com.crevion.apps.cleanandroidcode.models.CityListResponse;

/**
 * Created by yusufaw on 1/27/18.
 */

public interface HomeView {
    void showWait();

    void removeWait();

    void onFailure(String appErrorMessage);

    void geCityListSuccess(CityListResponse cityListResponse);
}

File: app/src/main/java/com/crevion/apps/cleanandroidcode/ui/home/HomeView.kt
package com.crevion.apps.cleanandroidcode.ui.home

import com.crevion.apps.cleanandroidcode.models.CityListResponse

/**
 * Created by yusufaw on 2/9/18.
 */

interface HomeView {
    fun showWait()

    fun removeWait()

    fun onFailure(appErrorMessage: String)

    fun geCityListSuccess(cityListResponse: CityListResponse)
}